src/Azs.Fabric.Admin/custom/Add-AzsScaleUnitNode.ps1

```diff
@@ -0,0 +1,191 @@
+#>
+function Add-AzsScaleUnitNode {
+[OutputType([System.Boolean])]
+[CmdletBinding(DefaultParameterSetName='ScaleExpanded', PositionalBinding=$false, SupportsShouldProcess, ConfirmImpact='Medium')]
+param(
+    [Parameter(ParameterSetName='Scale')]
+    [Parameter(ParameterSetName='ScaleExpanded')]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Category('Path')]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Runtime.DefaultInfo(Script='(Get-AzLocation)[0].Location')]
+    [System.String]
+    # Location of the resource.
+    ${Location},
+
+    [Parameter(ParameterSetName='Scale')]
+    [Parameter(ParameterSetName='ScaleExpanded')]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Category('Path')]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Runtime.DefaultInfo(Script='-join("System.",(Get-AzLocation)[0].Location)')]
+    [System.String]
+    # Name of the resource group.
+    ${ResourceGroupName},
+
+    [Parameter(ParameterSetName='Scale', Mandatory)]
+    [Parameter(ParameterSetName='ScaleExpanded', Mandatory)]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Category('Path')]
+    [System.String]
+    # Name of the scale units.
+    ${ScaleUnit},
+
+    [Parameter(ParameterSetName='Scale')]
+    [Parameter(ParameterSetName='ScaleExpanded')]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Category('Path')]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Runtime.DefaultInfo(Script='(Get-AzContext).Subscription.Id')]
+    [System.String]
+    # Subscription credentials that uniquely identify Microsoft Azure subscription.
+    # The subscription ID forms part of the URI for every service call.
+    ${SubscriptionId},
+
+    [Parameter(ParameterSetName='ScaleViaIdentity', Mandatory, ValueFromPipeline)]
+    [Parameter(ParameterSetName='ScaleViaIdentityExpanded', Mandatory, ValueFromPipeline)]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Category('Path')]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Models.IFabricAdminIdentity]
+    # Identity Parameter
+    # To construct, see NOTES section for INPUTOBJECT properties and create a hash table.
+    ${InputObject},
+
+    [Parameter(ParameterSetName='ScaleExpanded')]
+    [Parameter(ParameterSetName='ScaleViaIdentityExpanded')]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Category('Body')]
+    [System.Management.Automation.SwitchParameter]
+    # Flag indicates if the operation should wait for storage to converge before returning.
+    ${AwaitStorageConvergence},
+
+	[Parameter(ParameterSetName='BmciPv4Address', Mandatory)]
+    [Parameter(ParameterSetName='ScaleExpanded', Mandatory)]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Category('Path')]
+    [System.String]
+    # BMC address of the physical machine.
+    ${BmciPv4Address},
+
+	[Parameter(ParameterSetName='ComputerName ', Mandatory)]
+    [Parameter(ParameterSetName='ScaleExpanded', Mandatory)]
```

Comment:
ComputerName  and BmciPv4Address parameters are in two different parameter set, i think you want them to be mandatory in the same parameter set

Reply:
"ScaleExpanded" is the right parameter set for both of them, good catch!